src/slurm_plugin/clustermgtd.py

```diff
@@ -440,47 +419,35 @@ def manage_cluster(self):
             ComputeFleetStatus.PROTECTED,
         }:
             # Get node states for nodes in inactive and active partitions
+            # Initialize nodes
             try:
                 log.info("Retrieving nodes info from the scheduler")
-                partitions = {
-                    partition.name: partition for partition in ClusterManager._get_partition_info_with_retry()
-                }
-                log.debug("Partitions: %s", partitions)
-                active_nodes, inactive_nodes = self._get_node_info_from_partition(partitions)
-                log.debug("Current active slurm nodes in scheduler: %s", active_nodes)
-                log.debug("Current inactive slurm nodes in scheduler: %s", inactive_nodes)
-                self._inactive_partitions = get_inactive_partition_names(partitions)
+                nodes = self._get_node_info_with_retry()
+                log.debug("Nodes: %s", nodes)
+                partitions_name_map = self._retrieve_scheduler_partitions(nodes)
             except ClusterManager.SchedulerUnavailable:
```

Comment:
here we are handling SchedulerUnavailable exception but _get_node_info_with_retry and _retrieve_scheduler_partitions are not throwing that error in case of failure. We should fix that and make sure we properly handle errors and do not print stack trace in logs.

Reply:
Change to raise exception in _retrieve_scheduler_partitions, then handle Exception here to cover exception from _get_node_info_with_retry

Reply:
I think we should be consistent here: either we have both _get_node_info_with_retry and _retrieve_scheduler_partitions raise a SchedulerUnavailable when the command fails or we always propagate an exception and get rid of `ClusterManager.SchedulerUnavailable`.

Reply:
Done by get rid of ClusterManager.SchedulerUnavailable and use Exception